public/javascripts/common/wrapperFrame.js

```diff
@@ -0,0 +1,101 @@
+const iframe = document.getElementById('wrapper-frame');
+
+// Get the element for the dev enviroment warning.
+const testUserBanner = document.querySelector('.test-server-banner');
+
+// Define a function 'iframeURLChange' that takes a callback to execute when the iframe's URL changes.
+function iframeURLChange(callback) {
+    
+    // Define a function 'unloadHandler' that will be called when the iframe's content is unloaded.
+    var unloadHandler = function () {
+        
+        // Hide the iframe by setting its display to "none".
+        iframe.style.display = "none";
+        
+        // Call the callback with the new iframe URL after a brief delay (0ms, to ensure execution timing).
+        setTimeout(() => {
+            callback(iframe.contentWindow.location.href);
+        }, 0);
+    };
+
+    // Define a function 'attachUnload' to add or replace the 'unload' event listener on the iframe's content window.
+    function attachUnload() {
+        // First, remove any existing 'unload' event listener to prevent duplication.
+        iframe.contentWindow.removeEventListener("unload", unloadHandler);
+        
+        // Add the 'unload' event listener to the iframe's content window to call 'unloadHandler' when the content unloads.
+        iframe.contentWindow.addEventListener("unload", unloadHandler);
+    }
+
+    // Attach the unload event when the iframe finishes loading.
+    iframe.addEventListener("load", attachUnload);
+    
+    // Immediately call 'attachUnload' in case the content has already loaded.
+    attachUnload();
+}
+
+// Define a function 'scaleIframeContent' to adjust the iframe content's scale to fit within the window.
+function scaleIframeContent() {
+    // Set the device pixel ratio to 1 to avoid pixelation issues with Google Street View.
+    iframe.contentWindow.devicePixelRatio = 1;
+    
+    // Access the document inside the iframe.
+    const iframeDocument = iframe.contentDocument || iframe.contentWindow.document;
+    
+    // Find the element inside the iframe with the class 'tool-ui', which is the content to be scaled.
+    const contentElement = iframeDocument.querySelector('.tool-ui');
+
+    // Check if the content element exists before proceeding.
+    if (contentElement) {
+        
+        // Get the width and height of the window (excluding 70px for fixed elements like navbar or the dev env warning).
+        const iframeWidth = window.innerWidth;
+        const iframeHeight = window.innerHeight - 70 - (testUserBanner ? testUserBanner.clientHeight : 0);
+        
+        // Get the width and height of the content inside the iframe.
+        const contentWidth = contentElement.clientWidth;
+        const contentHeight = contentElement.clientHeight + 60; // Add 100px for extra padding.
```

Comment:
Hmm, if I remove the +100 then part of the UI is cut off on the bottom. Not a big deal because we can play around with the values to get what we want, but ideally the math here would work as we expect!
![Screenshot from 2024-09-30 14-34-43](https://github.com/user-attachments/assets/dd1fe4eb-b3ed-4824-bc4d-bb1e676e33a7)

I assume that part of this has to do with how we are adding padding-top to the body element? But not totally sure. Would be nice to have the padding-top make some sense, as I'm not really sure how it works or how it is supposed to right now!

I think that in an ideal world, we would be able to specify what percentage of the area below the navbar is to be used as white space both above and below the UI... It's not really _that_ important to be able to do this, but whenever I'm fully invested in a section of the code like this, I like to try to refactor a bit to make things make more sense for the future. So if it doesn't feel like a huge lift, any movement in that direction would be great to see!

Reply:
Agreed, I made those changes now!